# Модуль отладки

## Обзор

Модуль содержит функции для отладки и логирования сообщений, ошибок, а также для управления версиями.

## Подробней

Этот модуль предоставляет простой способ логирования информации и ошибок во время выполнения программы. Он позволяет включать и выключать логирование, устанавливать обработчик логов, а также хранить логи в списке.

## Функции

### `log`

```python
def log(*text: Any, file: Optional[Any] = None) -> None:
    """Log a message if logging is enabled."""
```

**Назначение**: Регистрирует сообщение, если включено логирование.

**Параметры**:

- `*text` (Any): Переменное количество аргументов, представляющих текст для логирования.
- `file` (Optional[Any], optional): Необязательный аргумент, указывающий файл для записи логов. По умолчанию `None`.

**Возвращает**:

- `None`: Функция ничего не возвращает.

**Как работает функция**:

1. Проверяет, включено ли логирование (переменная `logging` имеет значение `True`).
2. Если логирование включено, вызывает обработчик логов (`log_handler`) с переданными текстовыми сообщениями и файлом (если указан).

```ascii
Логирование включено? -- Да --> Вызов обработчика логов
|
Нет --> Конец
```

**Примеры**:

```python
logging = True
log_handler = print
log("Сообщение для логирования")
```

### `error`

```python
def error(*error: Any, name: Optional[str] = None) -> None:
    """Log an error message to stderr."""
```

**Назначение**: Регистрирует сообщение об ошибке в stderr.

**Параметры**:

- `*error` (Any): Переменное количество аргументов, представляющих сообщения об ошибках.
- `name` (Optional[str], optional): Необязательное имя ошибки. По умолчанию `None`.

**Возвращает**:

- `None`: Функция ничего не возвращает.

**Как работает функция**:

1. Преобразует каждое сообщение об ошибке в строку, добавляя имя типа ошибки, если предоставлено имя.
2. Вызывает функцию `log` для записи преобразованных сообщений об ошибках в stderr.

```ascii
Преобразование ошибок --> Вызов логгера для stderr
```

**Примеры**:

```python
error("Произошла ошибка", name="MyError")